packages/zoe/src/internal-types.js

```diff
@@ -223,19 +226,18 @@
 /**
  * @typedef {Object} AdminNode
  * A powerful object that can be used to terminate the vat in which a contract
- * is running, to get statistics, or to be notified when it terminates. The
- * object is only available from within the contract so
- * that clients of the contract can tell (by getting the source code from Zoe
- * using the installation) what use the contract makes of it. If they want to
- * be assured of discretion, or want to know that the contract doesn't have the
- * ability to call terminate(), Zoe makes this visible.
+ * is running, to get statistics, or to be notified when it terminates.
  *
  * @property {() => Promise<void>} done
- * provides a promise that will be fulfilled when the contract is terminated.
- * @property {() => void} terminate
- * kills the vat in which the contract is running
+ * returns a promise that will be fulfilled or rejected when the contract is
+ * terminated. If the contract terminates with a failure, the promise will be
+ * rejected with the reason. If the contract terminates successfully, the
+ * promise will fulfill to the completion value.
+ * @property {(completion: string) => void} terminate
+ * Terminate the vat in which the contract is running as a failure. This
+ * bypasses notification of Zoe.
```

Comment:
```suggestion
 * @property {(reason: Error) => void} terminateWithFailure
 * Terminate the vat in which the contract is running as a failure. This
 * bypasses notification of Zoe.
```

Reply:
Hang on, how is payout liveness upheld if we can terminate without notifying Zoe?